### PR TITLE
Remove fetching namescpaceURL from FeatureWFSTRequestBuilder and inst…

### PR DESCRIPTION
**Pull request by osqq**
…ead handle namespace in InsertFeatureHandler

- remove layer's namespace from InsertFeature

- take namespace name from potential prefix and add it to layerURL

### CODE REVIEW

**Comment by ZakarFin** [comment]
Because of oskariorg/oskari-docs#169 this needs to be modified so that the setters on OskariLayer are not called.

File: control-base/src/main/java/fi/nls/oskari/control/feature/AbstractFeatureHandler.java [hunk 1]
```diff
@@ -264,4 +269,25 @@ protected void hasUserPermissionEditLayers(Map<Integer, OskariLayer> layers, Use
             }
         }
     }
+
+    /**
+     * Takes workspace prefix from layer name (before ':') and puts it into the layer url after 'geoserver/'
+     * NOTE! May not work with other than geoserver
+     * @param  layer  OskariLayer layer
+     */
+    protected String movePrefixFromNameToURL(String layerName, String url){
```
Comment (ZakarFin) [inline control-base/src/main/java/fi/nls/oskari/control/feature/AbstractFeatureHandler.java]: Rename this to something like getURLForNamespace() or something like that.

File: control-base/src/main/java/fi/nls/oskari/control/feature/AbstractFeatureHandler.java [hunk 2]
```diff
@@ -77,9 +77,9 @@ protected int getLayerId(String layerId) throws ActionParamsException {
         return id;
     }
 
-    protected String postPayload(OskariLayer layer, String payload) throws ActionException {
+    protected String postPayload(OskariLayer layer, String payload, String url) throws ActionException {
```
Comment (ZakarFin) [inline control-base/src/main/java/fi/nls/oskari/control/feature/AbstractFeatureHandler.java]: Remove the OskariLayer from params as it's only needed for username/password. Add user/pass to params instead.

File: control-base/src/main/java/fi/nls/oskari/control/feature/DeleteFeatureHandler.java [hunk 3]
```diff
@@ -36,7 +36,10 @@ public void handlePost(ActionParameters params) throws ActionException {
 
         try {
             String payload = createPayload(jsonObject);
-            String responseString = postPayload(layer, payload);
+            String layerName = layer.getName();
+            String url = layer.getUrl();
```
Comment (ZakarFin) [inline control-base/src/main/java/fi/nls/oskari/control/feature/DeleteFeatureHandler.java]: Maybe take out a few lines with:

    String url = movePrefixFromNameToURL(layerName, layer.getUrl());

Or even

    String responseString = postPayload(layer, payload, movePrefixFromNameToURL(layerName, layer.getUrl()));

File: control-base/src/main/java/fi/nls/oskari/control/feature/AbstractFeatureHandler.java [hunk 4]
```diff
@@ -107,9 +107,14 @@ protected Feature getFeature(JSONObject jsonObject) throws ActionParamsException
         CoordinateReferenceSystem crs = CRS.decode(srsName);
         WFSLayerAttributes attrs = new WFSLayerAttributes(layer.getAttributes());
         WFSLayerCapabilities caps = new WFSLayerCapabilities(layer.getCapabilities());
+        String layerName = layer.getName();
+        //remove prefix from layername
+        if(layerName.indexOf(":") != -1){
+            layerName = layerName.substring(layerName.indexOf(":")+1);
+            layerName.trim();
```
Comment (jampukka) [inline control-base/src/main/java/fi/nls/oskari/control/feature/AbstractFeatureHandler.java]: The String returned by trim() is ignored, this line does nothing

**Pull request merged**